backend/lcfs/utils/spreadsheet_builder.py

```diff
@@ -0,0 +1,143 @@
+from io import BytesIO
+import logging
+import pandas as pd
+from openpyxl import styles
+import xlwt
+
+class SpreadsheetBuilder:
+    """
+    A class to build spreadsheets in xlsx, xls, or csv format.
+    Allows adding multiple sheets with custom styling and exporting them as a byte stream.
+
+    Example:
+    --------
+    builder = SpreadsheetBuilder('xlsx') # xlsx, xls, csv
+
+    # First sheet
+    builder.add_sheet(
+        'Employees', 
+        ['Name', 'Dept'], # Columns
+        [['Alex', 'HR'], ['Mike', 'IT']], # Rows
+        styles={'bold_headers': True, 'column_widths': [10, 20]}
+    )
+
+    # Second sheet
+    builder.add_sheet( ... )
+
+    file_content = builder.build_spreadsheet()
+    """
+    def __init__(self, file_format: str = "xls"):
+        self.file_format = file_format
```

Comment:
Could we add the file format validation to the __init__ method to catch this earlier?

Reply:
Of course! Thank you for pointing this out.